components/cosmetic_filters/resources/data/procedural_filters.ts

```diff
@@ -259,6 +259,13 @@ const operatorCssSelector = (selector: CSSSelector,
 
   if (element.matches(selector)) {
     return [element]
+  } else {
+    const allChildNodes = _allChildren(element);
+    for (const aNode of allChildNodes) {
+      if (aNode.matches(selector)) {
+        return [aNode]
+      }
+    }
```

Comment:
the `if`/`else` may be problematic in rare cases here, we should be selecting _all_ matching elements

Reply:
Ah ok may need to filter the `allChildNodes` array instead.

Reply:
also, maybe I'm mistaken, but I feel like there ought to be a way to do this purely with CSS?

Reply:
As discussed I've updated to including the child nodes in `onMutations` so the child nodes are all checked against each case in `operatorCssSelector` in [7157e69](https://github.com/brave/brave-core/pull/29139/commits/7157e69ca63d2db17e6ad6f298480207ce790f2a).

The code is similar on iOS & desktop/android, but I think best to split that refactor work out of this bugfix to a separate sub-issue of https://github.com/brave/brave-browser/issues/41017.